Issue 19 - call to undefined method WikiPage::factory()

gesinn-it-ilm opened the pull request:
This PR is related to the issue #19. 

This PR contains:

- method `isUniqueId()` updated, renamed to `getUniqueIdChecker()`
- increase codecov for `IdProviderFactory.php`, cover `getUniqueIdChecker()` function

gesinn-it-ilm committed:
IdProviderFactory.php

- update isUniqueId()
- increase test coverage

gesinn-it-ilm committed:
IdProviderFactory.php

- update isUniqueId()
- update tests

gesinn-it-ilm commented:
@gesinn-it-gea  could you please review this and if this is ok, we can merge this one as well.

gesinn-it-gea commented on src/IdProviderFactory.php:
```diff
@@ -59,13 +59,20 @@ private static function dbExecute() {
+	 * @param string $text The title or ID to check for uniqueness of the wiki page.
 	 * @return \Closure
 	 */
 	private static function isUniqueId() {
```
Naming:

The method name, isUniqueId, suggests it checks uniqueness directly, but the implementation returns a closure. Renaming it to something like getUniqueIdChecker might better reflect its functionality.

gesinn-it-gea commented on src/IdProviderFactory.php:
```diff
@@ -59,13 +59,20 @@ private static function dbExecute() {
-		return function ( $id ) {
-			$title = Title::newFromText( $id );
+		return function ( $text ) {
+			$title = Title::newFromText( $text );
```
Error Handling:

The code does not handle cases where `Title::newFromText` or the page factory methods might throw exceptions or return unexpected results. Adding error handling could make the function more robust. We should either add local exception handling or ensure, more global exception handling exists. At least we should have a test to feed some non-existing Title to see what happens. Do we have one?

```
try {
    $title = Title::newFromText($text);
    if ($title === null) {
        return true;
    }

    if (method_exists(MediaWikiServices::class, 'getWikiPageFactory')) {
        $page = MediaWikiServices::getInstance()->getWikiPageFactory()->newFromTitle($title);
    } else {
        $page = WikiPage::factory($title);
    }

    return !$page->exists();
} catch (Exception $e) {
    // Log or handle the error as necessary
    return false;
}
```

gesinn-it-gea commented on src/IdProviderFactory.php:
```diff
@@ -59,13 +59,20 @@ private static function dbExecute() {
 
 	/**
-	 * Checks whether a WikiPage with the following id/title already exists
+	 * Function to check if string id/title already exists, i.e. is already used as a
```
Documentation:

The PHPDoc is clear but could mention that the method returns a closure and explain why this design choice was made. For example:

gesinn-it-gea reviewed (commented):
see inline comments

gesinn-it-ilm committed:
IdProviderFactory.php

- new method naming
- instead of isUniqueId use name getUniqueIdChecker to reflect the functionality better

gesinn-it-ilm committed:
getUniqueIdChecker()

- update PHPDoc

gesinn-it-ilm committed:
IdProviderFactoryTest.php

- add new test cases to check whenever the page exists or not

gesinn-it-ilm committed:
IdProviderFactory.php

- update function param, add $id instead of $text

gesinn-it-ilm commented:
@gesinn-it-gea PR updated and ready to merge.
